src/command/backup/backup.c

```diff
@@ -0,0 +1,1911 @@
+        }
+    }
+    MEM_CONTEXT_TEMP_END();
+
+    FUNCTION_LOG_RETURN(BOOL, result);
+}
+
+/***********************************************************************************************************************************
+Start the backup
+***********************************************************************************************************************************/
+typedef struct BackupStartResult
+{
+    String *lsn;
+    String *walSegmentName;
+    VariantList *dbList;
+    VariantList *tablespaceList;
+} BackupStartResult;
+
+#define FUNCTION_LOG_BACKUP_START_RESULT_TYPE                                                                                      \
+    BackupStartResult
+#define FUNCTION_LOG_BACKUP_START_RESULT_FORMAT(value, buffer, bufferSize)                                                         \
+    objToLog(&value, "BackupStartResult", buffer, bufferSize)
+
+static BackupStartResult
+backupStart(BackupData *backupData)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(BACKUP_DATA, backupData);
+    FUNCTION_LOG_END();
+
+    BackupStartResult result = {.lsn = NULL};
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        // If this is an offline backup
+        if (!cfgOptionBool(cfgOptOnline))
+        {
+            // Check if Postgres is running and if so only continue when forced
+            if (storageExistsP(backupData->storagePrimary, PG_FILE_POSTMASTERPID_STR))
+            {
+                if (cfgOptionBool(cfgOptForce))
+                {
+                    LOG_WARN(
+                        "--no-" CFGOPT_ONLINE " passed and " PG_FILE_POSTMASTERPID " exists but --" CFGOPT_FORCE " was passed so"
+                        " backup will continue though it looks like the postmaster is running and the backup will probably not be"
+                        " consistent");
+                }
+                else
+                {
+                    THROW(
+                        PostmasterRunningError,
+                        "--no-" CFGOPT_ONLINE " passed but " PG_FILE_POSTMASTERPID " exists - looks like the postmaster is running."
+                        " Shutdown the postmaster and try again, or use --force.");
+                }
+            }
+        }
+        // Else start the backup normally
+        else
+        {
+            // Check database configuration
+            checkDbConfig(backupData->version, backupData->pgIdPrimary, backupData->dbPrimary, false);
+
+            // Start backup
+            LOG_INFO_FMT(
+                "execute %sexclusive pg_start_backup(): backup begins after the %s checkpoint completes",
+                backupData->version >= PG_VERSION_96 ? "non-" : "",
+                cfgOptionBool(cfgOptStartFast) ? "requested immediate" : "next regular");
+
+            DbBackupStartResult dbBackupStartResult = dbBackupStart(
```

Comment:
1a) Weird -- what test are you running to get this error?  I can't reproduce it.  I think the error may be in a previous statement.
2b) I'm not worried about these times being synced since this time is not used for anything.
3) Yes -- backup.c, line 225.

Reply:
1a) the error was my fault so ignore.
2b) I'd like to understand more of how we are using the timestamps now so let's discuss
3) OK